### PR TITLE
Add sleep timeout as a fallback to prevent deadlocks

### PR DESCRIPTION
**Pull request by derekjn**

### CODE REVIEW

File: src/backend/pipeline/tuplebuf.c [hunk 1]
```diff
@@ -614,7 +614,7 @@ TupleBufferTryWait(TupleBufferReader *reader)
 	SpinLockRelease(&buf->mutex);
 
 	if (should_wait)
-		WaitLatch(reader->proc->latch, WL_LATCH_SET | WL_POSTMASTER_DEATH, 0);
+		WaitLatch(reader->proc->latch, WL_LATCH_SET | WL_POSTMASTER_DEATH, 1);
```
Comment (usmanm) [inline src/backend/pipeline/tuplebuf.c]: So the problem was that the combiners weren't waking up to read from the CombinerTupleBuffer?

Comment (derekjn) [inline src/backend/pipeline/tuplebuf.c]: Correct.

File: src/backend/pipeline/cont_scheduler.c [hunk 2]
```diff
@@ -673,6 +673,10 @@ ContQuerySchedulerMain(int argc, char *argv[])
 	ContQuerySchedulerShmem->scheduler_pid = MyProcPid;
 
 	dbs = get_database_list();
+	if (list_length(dbs) * TOTAL_SLOTS > max_worker_processes)
```
Comment (usmanm) [inline src/backend/pipeline/cont_scheduler.c]: We should probably throw an error and not start any continuous query procs. Seeing the state where one db gets all the workers and the other one doesn't seems like something we shouldn't do.

File: src/backend/pipeline/tuplebuf.c (hunk 1 above)
Comment (usmanm) [inline src/backend/pipeline/tuplebuf.c]: This fix might not be too great then because this means that our background processes will never go to sleep.

Comment (derekjn) [inline src/backend/pipeline/tuplebuf.c]: We'll they'll sleep for at least 1ms, so when they're not busy they'll spend the majority of their time asleep right?

Comment (usmanm) [inline src/backend/pipeline/tuplebuf.c]: I guess, it's basically going be like having `list_length(db) * TOTAL_SLOTS* processes in a loop like:
```
while (TupleBufferIsEmpty(buf))
  pg_usleep(1000);
```

Comment (derekjn) [inline src/backend/pipeline/tuplebuf.c]: I just ran that loop with 32 procs and the cpu usage is negligible. How about we increase to 10ms and only wait on combiners?

Comment (usmanm) [inline src/backend/pipeline/tuplebuf.c]: Oh, in that case, keep it at 1ms. I forget how fast computers are.

Comment (derekjn) [inline src/backend/pipeline/tuplebuf.c]: I'll leave it at 10 and only on combiners, just to be sure. It's a fallback for a case that shouldn't be happening anyways...

**Pull request closed without merging**